src/coprocessor/dag/rpn_expr/impl_math.rs

```diff
@@ -0,0 +1,126 @@
+use cop_codegen::rpn_fn;
+
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::Error;
+use crate::coprocessor::Result;
+
+#[rpn_fn]
+#[inline]
+fn abs_int(arg: &Option<Int>) -> Result<Option<Int>> {
+    match arg {
+        None => Ok(None),
+        Some(arg) => match (*arg).checked_abs() {
+            None => Err(Error::overflow("BIGINT", &format!("abs({})", *arg)))?,
+            Some(arg_abs) => Ok(Some(arg_abs)),
+        },
+    }
+}
+
+#[rpn_fn]
+#[inline]
+fn abs_uint(arg: &Option<Int>) -> Result<Option<Int>> {
+    match arg {
+        Some(arg) => Ok(Some(*arg)),
+        None => Ok(None),
+    }
+}
+
+#[rpn_fn]
+#[inline]
+fn abs_real(arg: &Option<Real>) -> Result<Option<Real>> {
+    // abs returns NAN if the number is NAN, so don't worry about it
+    match arg {
+        Some(arg) => {
+            let f = arg.abs();
+            Ok(Some(Real::new(f).unwrap()))
+        }
+        None => Ok(None),
+    }
+}
+
+#[rpn_fn]
+#[inline]
+fn abs_decimal(arg: &Option<Decimal>) -> Result<Option<Decimal>> {
+    use crate::coprocessor::codec::mysql::Res;
+
+    match arg {
+        Some(arg) => match arg.to_owned().abs() {
+            Res::Ok(v) => Ok(Some(v)),
+            Res::Truncated(_) => Err(Error::truncated())?,
```

Comment:
Decimal's `Result` implements `Into` so that `Ok` becomes `Ok` and `Truncated` / `Overflow` becomes `Error`. I think we can utilize this trait here.

Reply:
You may refer to the original implementation of abs_decimal. It is not that complex.